frontend/ui/widget/buttondialog.lua

```diff
@@ -1,3 +1,73 @@
+--[[--
+A button dialog widget that shows a grid of buttons.
+
+    @usage
+    local button_dialog = ButtonDialog:new{
+        buttons = {
+            {
+                {
+                    text = "First row, left side",
+                    callback = function()
+
+                    end,
+                    hold_callback = function()
+
+                    end
```

Comment:
It does become rather large this way. Maybe just `callback = …` or is that unclear?

Reply:
I think that's A-OK

Reply:
That helps, but actually with or without hold_callback I meant to do one line instead of 3.

```suggestion
                    callback = function() end,
                    hold_callback = function() end
```